### ISSUE TITLE
Unnecessary compute() calls in the Trainer class lead to performance degradation for object detection tasks

### ISSUE THREAD

**Issue by AlfredQin** [Deci-AI/super-gradients#957]
###Description:

I have encountered a performance issue while using the Trainer class for object detection tasks. The issue arises from the way the Trainer class interacts with the Metric class of the torchmetric library.

###Current behavior:

Currently, during a mini-batch iteration, the Trainer class always calls the compute() method of the Metric class.
 ` logging_values = loss_avg_meter.average + get_metrics_results_tuple(self.train_metrics)
`
```
def get_metrics_results_tuple(metrics_collection: MetricCollection):
    """
    :param metrics_collection: metrics collection of the user specified metrics
    @type metrics_collection
    :return: tuple of metrics values
    """
    if metrics_collection is None:
        results_tuple = ()
    else:
        results_tuple = tuple(flatten_metrics_dict(metrics_collection.compute()).values())
    return results_tuple
```
Here is the code of my Metric class, which is inherited from torchmetrics.detection.MeanAveragePrecision
```
class MyMeanAveragePrecision(MeanAveragePrecision):
    def __init__(
            self,
            post_prediction_callback=None,
            box_format: str = "xyxy",
            iou_thresholds: Optional[List[float]] = None,
            rec_thresholds: Optional[List[float]] = None,
            max_detection_thresholds: Optional[List[int]] = None,
            class_metrics: bool = False,
            compute_on_step: Optional[bool] = None,
            **kwargs: Dict[str, Any],
    ):
        super().__init__(
            box_format=box_format,
            iou_thresholds=iou_thresholds,
            rec_thresholds=rec_thresholds,
            max_detection_thresholds=max_detection_thresholds,
            class_metrics=class_metrics,
            compute_on_step=compute_on_step,
            **kwargs,
        )
        self.post_prediction_callback = post_prediction_callback
        self.component_names = ['map', 'map_50', 'map_75']

    def update(self, preds, target):
        orig_target_sizes = torch.stack([t["orig_size"] for t in target], dim=0)
        results = self.post_prediction_callback['bbox'](preds, orig_target_sizes)
        super().update(results, target)

    def compute(self):
        metrics = {k: v for k, v in super().compute().items() if k in self.component_names}

        return metrics
```
This is inefficient for object detection tasks, where the mean average precision (mAP) is typically computed only after the completion of an entire epoch. As a result, the compute() method is invoked much more frequently than necessary - once for each mini-batch, rather than once per epoch.

###Impact:

This behavior significantly decreases training speed because the compute() method is computationally expensive, and it is being called many more times than it needs to be.

###Expected behavior:

For object detection tasks, the compute() method should only be called once per epoch, not once per mini-batch.

**Comment by shaydeci** [comment]
@AlfredQin thanks for pointing this out.
We will consider making it optional to see the metrics progress throughout the batch iterations (which is the root cause of callling compute).
